apps/jupyter/run-jupyter.sh

```diff
@@ -42,19 +51,20 @@ else
         2>> "${SSLLOG}" || die "Failed to create ${HOST} certificate"
         echo "Created ${HOST} certificate"
     fi
-    SSLARG="--certfile=\"${SSLPFX}-cert.pem\" --keyfile=\"${SSLPFX}-key.pem\""
+    SSLARG=("--certfile=\"${SSLPFX}-cert.pem\"" "--keyfile=\"${SSLPFX}-key.pem\"")
 fi
 
 (
-    while [[ $(ls /app/.local/share/jupyter/runtime/jpserver-*-open.html 2>/dev/null | wc -w) = 0 ]]; do
+    while [[ $(find "${RUNTIME_DIR}" -type f -name 'jpserver-*-open.html' 2>/dev/null | wc -l) = 0 ]]; do
         echo "Waiting for jpserver-*-open.html to appear"
         sleep 1
     done
-    FILE="$(ls /app/.local/share/jupyter/runtime/jpserver-*-open.html)"
-    if [[ $(echo "${FILE}" | wc -w) != 1 ]]; then
-        echo "ERROR: got '${FILE}' for jpserver files"
-        ls /app/.local/share/jupyter/runtime
-        echo "ERROR: Multiple jpsterver-*-open.html files"
+    FILE="$(find "${RUNTIME_DIR}" -type f -name 'jpserver-*-open.html')"
+    if [[ $(wc -l <<< ${FILE}) != 1 ]]; then
+        CONCAT=$(xargs -r <<< ${FILE})
+        echo "ERROR: got '${CONCAT}' for jpserver files"
+        ls "${RUNTIME_DIR}"
```

Comment:
find -ls
(to match with the uses of find elsewhere, which would also have issues if the files are in subdirectories while ls won't show them)

Reply:
We do print out the matches in the line above.  We could just use ls -R if we don't expect too much output.

Reply:
Why not ```find "${RUNTIME_DIR}" -type f -ls```
I'm also ok with ls -R, but since you switched to find everywhere else it seems better to be consistent.